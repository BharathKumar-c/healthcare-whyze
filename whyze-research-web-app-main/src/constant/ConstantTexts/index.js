import React from 'react';

// dashboard welcome Screen const text

export const welComeComponentContent =
  'See how Whyze is preforming for researchers like you.';
export const createFirstStudy = 'Create your first study';
export const createFirstStudyContent =
  'Define your protocol or create your first feasibility study';
export const patientstitleLable = 'Patients';
export const totalPatientsLable = 'Total patients';
export const countriesLable = 'Countries';
export const sitesLable = 'Sites';
export const sitesRecruitedLable = 'Sites recruited';
export const avgTimeLable = 'Avg. response time';
export const trialsLable = 'Trials';
export const activeTrials = 'Active trials';
export const totalTrials = 'Total trials';
export const screenFailRate = 'Screen fail rate';
export const threeMonths = '3 months';
export const avgTimeToRecruit = 'Avg. time to recruit';
export const feasibilityStudiesInPlanning = 'Feasibility studies in planning';
export const seeAll = 'See all';

//Profile Page const text
export const ResearcherTitle = 'Researcher title -';
export const AuthProfile = 'AUTHENTICATED PROFILE';
export const ContactDetails = 'Contact details';
export const CompanyEmail = 'Company email address';
export const PhoneNumber = 'Phone';

//ContactModal Page Const texts

export const EmailAddress = 'Email address';
export const Phone = 'Phone';
export const okButtonLabel = 'Ok';
export const saveButtonLabel = 'save';
export const cancelButtonLabel = 'cancel';
export const contactDetailsLabel = 'Contact details';

//Addcriteria texts
export const addCriteriaTitle = 'Select a criteria item';

//RenameModal texts
export const renameModalTitle = 'Profile details';
export const fullNameLabel = 'Full name';
export const titleLabel = 'Title';

// Medication constant texts
export const medicationConstants = {
  medicationLabel: 'Medication',
  medicationErrorMessage: 'Please add the Medication',
  selectlabel: 'Select Dosage',
  inputLabel: 'Enter value',
  startedPrefixText: 'Started at least',
};
// Indication constant texts
export const indicationConstants = {
  medicalConditionLabel: 'Medical condition',
  medicalConditionErrorMessage: 'Please add the Medical condition',
};

//Allergies constant texte
export const allergiesConstants = {
  allergiesLabel: 'Allergies criteria',
  allergiesPlaceholder: 'Allergies',
  allergiesErrorText: 'Please add the Allergies',
};
//Symptom constant texts

export const symptomConstats = {
  symptomLabel: 'Symptom criteria',
  symptomPlaceholder: 'Symptom',
};
//Project settingmodal Constants

export const projectSettingConstants = {
  addProjectError: 'Please add the project',
  addProjectLabel: 'Project',
  addClientNameError: 'Please add ClientName',
  rangeLabel: 'Select',
  addClientLabel: 'Client name',
  description: 'Description',
  sharingLabel: 'Sharing',
  shareBtnLabel: 'Share',
  standardsText: 'Standards',
  standardDescription: 'Standardising markers and value outputs',
  measurementText: 'Default measurement',
  clinicalTrialError: 'Please add the clinical trial',
  clinicalTrial: 'Clinical Trial',
  sponsorError: 'Please add the sponsor',
  sponsor: 'Sponsor',
  projectSettings: 'Project settings',
  descriptionError: 'Please add the description',
  standardsError: 'Please select the standards',
  noIndication: 'No Indication',
};
export const shareProjectConstants = {
  addPeoplePlaceHolder: 'Add people and groups',
  accessPeopleText: 'People with access',
};
export const selectedPeoplesConstants = {
  addPeople: 'Add people and groups',
  notifyPeopleLabel: 'Notify people',
  messageLabel: 'Message',
};
// Nicotine constant texts
export const nicotineConstants = {
  nicotineConditionErrorMessage: 'Please add the nicotine condition',
  nicotineLabel: 'Nicotine criteria',
};

// Create Account constant texts
export const createAccountContants = {
  createAccounTitle: (
    <>
      Set up your <br /> Whyze account
    </>
  ),
  setUpWhyzeAccount: 'Set up your Whyze account',
  nameLabel: 'Name',
  emailLabel: 'Company email',
  phoneLabel: 'Your phone number',
  passwordLabel: 'Create password',
  characterLengthLabel: 'Between 8 and 20 characters',
  numericLabel: 'At least one number',
  uppercaseLabel: 'At least one upper-case letter',
  lowercaseLabel: 'At least one lower-case letter',
  createAccountButtonLabel: 'Create account',
  agreeConditionText: 'By clicking the button above you agree to our',
  termsOfUseText: 'Terms of use',
  privacyNoticesText: 'Privacy policies',
  alreadyHaveAccountText: 'Already have an account?',
  loginText: 'Login',
  termsOfUseLink: 'https://www.whyzehealth.com/privacy-notice/',
  privacyNoticesLink: 'https://www.whyzehealth.com/terms',
};

//indication const

export const conditions = {
  started: 'started',
  dosage: 'dosage',
  courseOfTherapy: 'course of therapy',
  diagnosed: 'diagnosed',
  recovered: 'recovered',
  severity: 'severity',
};

//demographics const

export const demographicsConst = {
  age: 'age',
  min: 'min',
  max: 'max',
  gender: 'gender',
};

//sites const

export const siteConst = {
  siteInformationCosnt: {
    patientSegmentHeader: 'Patient segment',
    connectedWithSite: 'Connected with this site',
    additionalCatchmentAreas:
      'Additional Whyze patients in site catchment areas',
    sitePerformance: 'Site performance',
    recruitmentTimeline: 'Recruitment timeline',
    dropOutRate: 'Drop out rate',
    trackRecordOfDoingTrials: 'Track record of doing trials',
    completedInThePastyear: 'Completed in the past year',
    primaryInvestigator: 'Primary Investigator',
    administration: 'Administration',
    screenFailRate: 'Screen fail rate',
    industryAverage: 'Industry average',
    avgResponseTime: 'Avg. response time',
  },
};

//setupAccount const

export const setupAccountConst = {
  dontHaveaccount: 'Don’t have an account yet?',
  setUpAccount: 'Set up account',
};

//forgotpassword const

export const forgotpasswordConst = {
  WHYZE: 'WHYZE',
  forgotPassword: 'Forgot Password',
  resendEmail: 'Enter your email below and we will send you a reset email.',
  pleaseTry: 'Didn’t receive it ? Please try again',
  returnTo: 'return to',
  loginPage: 'login',
};

//weight const  texts

export const weightConst = {
  weightRange: 'Weight Range',
  weightMin: 'min',
  weightMax: 'max',
  weightType: 'number',
  weightmaxErrorText: 'add max weight',
  weightminErrorText: 'add min weight',
  weightFrom: 'From',
  weightTo: 'To',
};

export const SaveAndCloseModalConst = {
  SaveAndCloseModalTitle: 'Name before exiting',
  projectNameInputLabel: 'Project',
  projectNameInputPlaceholder: 'New patient segment',
  clientNameInputLabel: 'Client name',
  descriptionNameInputLabel: 'Description',
  patientSegmentInputLabel: 'Patient segment required',
};

// Proposed Site list constant texts

export const proposedSiteListConstants = {
  title: 'Proposed Site List',
  seeMore: 'See more',
  emptyContent:
    'To create your proposed list, drag & drop or add your selected sites here.',
};

// Surgery constant text
export const surgeryConstants = {
  surgeryInputLabel: 'Surgery criteria',
  surgeryInputErrorMessage: 'Please add the Surgery',
};

export const feasibilityStudy = {
  newFeasibilityStudy: 'new-feasibility-study',
};
