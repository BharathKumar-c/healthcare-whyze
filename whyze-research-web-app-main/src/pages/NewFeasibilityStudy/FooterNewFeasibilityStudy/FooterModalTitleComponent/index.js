import React from 'react';
import PropTypes from 'prop-types';
import { Image } from '../../../../components/basic';

export default function Title({ title, logo, className }) {
  return (
    <div className={className}>
      <Image
        src={logo}
        alt="modal-title-logo"
        width={40}
        height={40}
        preview={false}
      />
      <h3>{title}</h3>
    </div>
  );
}

Title.propTypes = {
  title: PropTypes.string,
  logo: PropTypes.string,
  isSubComp: PropTypes.bool,
  className: PropTypes.string,
  onBack: PropTypes.func,
};

Title.defalutProps = {
  logo: '',
  bgImage: '',
  isSubComp: false,
  className: '',
  onBack: () => {},
};
