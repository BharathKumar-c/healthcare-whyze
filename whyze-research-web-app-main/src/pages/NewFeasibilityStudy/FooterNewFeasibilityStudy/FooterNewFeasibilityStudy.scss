@import '../../../common_styles/common_variable';

.feasibility-study-footer-container {
    width: 100%;
    padding: 1rem;
    color: $primary-color;
    box-shadow: 5px 8px 50px 18px rgba(57, 45, 46, 0.12);
    justify-content: center;
    display: flex;
    flex-direction: row;
    align-items: center;

    button {
        margin-right: 2.5rem;
        border-radius: 1.5rem;
        font-size: 15px;
        font-weight: 500;
        text-transform: capitalize;
        padding: 0rem 2rem 0rem 2rem;

        &:nth-child(1) {
            color: $primary-color;
            border: none;
            font-weight: 700;
            font-size: 16;
        }

        &:nth-child(2) {
            background: white;
            color: $primary-color;
            border-width: 1px;
            border-color: $primary-color;
            font-weight: 700;
            font-size: 16;
        }

        &:nth-child(3) {
            background: $primary-color;
            color: white;
            border: none;
            width: 150px;
            font-weight: 700;
            font-size: 16;
        }
    }
}

.save-and-close-modal-title{
    justify-content: space-between;
    display: flex;
    flex-direction: column;
    align-items: center;
    background-image: url("../../../assets/BackgroundShape.svg");
    background-repeat: no-repeat;
    background-position-y:1rem;
    background-position-x: center;
    height: 100%;
    padding-bottom:3%;
}

.save-and-close-modal-children{
    input{
        margin-bottom: 5%;
    }
    .empty-tag{
        margin-bottom: 5%;
    }
}