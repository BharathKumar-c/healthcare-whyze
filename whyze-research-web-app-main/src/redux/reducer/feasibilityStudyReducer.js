import { createSlice } from '@reduxjs/toolkit';
import { v4 as uuid } from 'uuid';

export const initialState = {
  segments: [
    {
      name: 'Segment-1',
      isDone: false,
      activeTab: {
        name: 'feasibility',
        index: 0,
      },
      tabs: [
        {
          name: 'feasibility',
          title: 'Feasibility',
          isActive: true,
          position: 'left',
        },
      ],
      feasibilityStudy: [
        {
          name: 'Indication',
          case: 'Indication',
          condition: 'Inclusion',
          caseName: '',
          child: [],
        },
        {
          name: 'Demographics',
          case: 'Demographics',
          condition: 'Inclusion',
          caseName: '',
          child: [
            {
              id: uuid().slice(0, 8),
              condition: 'Gender',
              field: 'gender',
              gender: '',
            },
            {
              id: uuid().slice(0, 8),
              condition: 'Age range',
              field: 'age',
              ageRange: {
                min: 0,
                max: 0,
              },
            },
          ],
        },
      ],
      country: [],
      location: [],
      selectedCountries: [],
      totalConnectedNotConnected: {},
      sites: [],
      sitesContacted: [],
      sitesInitiated: [],
      preferredSites: [],
      patientCount: {
        feasibility: 0,
        location: 0,
      },
      siteCount: {
        feasibility: 0,
        location: 0,
      },
    },
  ],
  cardValue: {
    case: 'Indication',
    condition: '',
    index: 0,
  },
  activeSegment: {
    name: '',
    index: 0,
  },
};

const feasibilityStudySlice = createSlice({
  name: 'feasibilityStudyAction',
  initialState: initialState,
  reducers: {
    setCardDetails: (state, { payload }) => {
      state.cardValue = payload;
    },
    setSegments: (state, { payload }) => {
      state.segments = payload;
    },
    setActiveSegment: (state, { payload }) => {
      state.activeSegment = payload;
    },
    resetSegments: (state) => {
      state.segments = initialState.segments;
      state.cardValue = {
        case: 'Indication',
        condition: '',
        index: 0,
      };
      state.activeSegment = {
        name: '',
        index: 0,
      };
    },
  },
});

export const { setCardDetails, setSegments, setActiveSegment, resetSegments } =
  feasibilityStudySlice.actions;

export default feasibilityStudySlice.reducer;
