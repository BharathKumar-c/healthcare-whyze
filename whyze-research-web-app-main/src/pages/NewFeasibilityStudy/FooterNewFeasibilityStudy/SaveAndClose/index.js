import React from 'react';
import PropTypes from 'prop-types';
import { Input, TextArea } from '../../../../components/basic';
import { Modal } from '../../../../components/hoc';
import WhyzeLogo from '../../../../assets/WhyzeLogo.svg';
import './../FooterNewFeasibilityStudy.scss';
import Title from '../FooterModalTitleComponent';
import {
  cancelButtonLabel,
  SaveAndCloseModalConst,
  saveButtonLabel,
} from '../../../../constant/ConstantTexts';

export default function SaveAndCloseModal({
  openModal,
  handleClose,
  handleSave,
  handleInputChange,
  errors,
  loading,
}) {
  return (
    <Modal
      className="save-and-close-modal"
      isOpen={openModal}
      onClose={handleClose}
      closable
      confirmLoading={loading}
      title={
        <Title
          title={SaveAndCloseModalConst.SaveAndCloseModalTitle}
          logo={WhyzeLogo}
          className="save-and-close-modal-title"
        />
      }
      width={450}
      handleActionClick={handleSave}
      okButtonLabel={saveButtonLabel}
      cancelButtonLabel={cancelButtonLabel}
    >
      <div className="save-and-close-modal-children">
        <Input
          id="main_textbox"
          lable={SaveAndCloseModalConst.projectNameInputLabel}
          name={'projectName'}
          placeholder={SaveAndCloseModalConst.projectNameInputPlaceholder}
          handleInputChange={(e) => handleInputChange(e)}
          isError={errors?.key === 'projectName' ? true : false}
          errorText={errors?.errors}
        />
        <Input
          id="main_textbox"
          lable={SaveAndCloseModalConst.clientNameInputLabel}
          name={'clientName'}
          placeholder={SaveAndCloseModalConst.clientNameInputLabel}
          handleInputChange={(e) => handleInputChange(e)}
          isError={errors?.key === 'clientName' ? true : false}
          errorText={errors?.errors}
        />
        <TextArea
          id="main_textbox"
          lable={SaveAndCloseModalConst.descriptionNameInputLabel}
          maxLength={200}
          name={'description'}
          placeholder={SaveAndCloseModalConst.descriptionNameInputLabel}
          handleInputChange={(e) => handleInputChange(e)}
          style={{ resize: 'none' }}
          isError={errors?.key === 'description' ? true : false}
          errorText={errors?.errors}
        />
        <div className="empty-tag" />
        <Input
          id="main_textbox"
          lable={SaveAndCloseModalConst.patientSegmentInputLabel}
          name={'patientSegmentReqiured'}
          placeholder={SaveAndCloseModalConst.patientSegmentInputLabel}
          handleInputChange={(e) => handleInputChange(e)}
          type={'number'}
          isError={errors?.key === 'patientSegmentReqiured' ? true : false}
          errorText={errors?.errors}
        />
      </div>
    </Modal>
  );
}

SaveAndCloseModal.propTypes = {
  openModal: PropTypes.bool,
  handleClose: PropTypes.func,
  handleSave: PropTypes.func,
  handleInputChange: PropTypes.func,
  isError: PropTypes.bool,
  loading: PropTypes.bool,
  errors: PropTypes.object,
};

SaveAndCloseModal.defalutProps = {
  openModal: false,
  handleClose: () => {},
  handleSave: () => {},
  handleInputChange: () => {},
  isError: false,
  loading: false,
  errors: {},
};
