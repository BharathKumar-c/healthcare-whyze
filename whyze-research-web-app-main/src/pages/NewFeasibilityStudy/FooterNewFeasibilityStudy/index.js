import React from 'react';
import './FooterNewFeasibilityStudy.scss';
import { Button } from '../../../components/basic';
import { useSelector, useDispatch } from 'react-redux';
import { setSegments } from '../../../redux/reducer/feasibilityStudyReducer';
import useSaveandCloseService from './SaveAndClose/useSaveandCloseService';
import { useNavigate } from 'react-router-dom';
import { RoutesUrls } from '../../../constant/RoutesUrls';
import { clearSettings } from '../../../redux/reducer/projectsReducer';
import { resetSegments } from '../../../redux/reducer/feasibilityStudyReducer';
import { filterSelectedCountries } from '../../../utils/Formatter';

const FooterNewFeasibilityStudy = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const { saveUserProject } = useSaveandCloseService();
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const newTabs = segments[activeSegment.index]?.tabs;
  const { sites, activeTab, patientCount, selectedCountries } =
    segments[activeSegment.index];

  const handleSave = () => {
    saveUserProject();
    dispatch(clearSettings());
    dispatch(resetSegments());
  };

  const handleCancel = () => {
    navigate(`/${RoutesUrls.dashBoardUrl}`);
    dispatch(resetSegments());
    dispatch(clearSettings());
  };

  const handleNext = () => {
    const data = JSON.parse(JSON.stringify(newTabs));
    const newSegments = JSON.parse(JSON.stringify(segments));
    if (newSegments[activeSegment?.index].activeTab?.name === 'feasibility') {
      const { location, selectedCountries } = newSegments[activeSegment?.index];
      if (
        selectedCountries &&
        selectedCountries?.length > 0 &&
        location &&
        location?.length > 0
      ) {
        const resetedSelectedCountries = filterSelectedCountries(
          location,
          selectedCountries,
        );

        newSegments[activeSegment?.index].selectedCountries =
          resetedSelectedCountries;
      }
    }

    for (var i in data) {
      if (data[i].isActive) {
        data[i].isActive = false;
        data[i].position = 'left';
      }
      if (newSegments[activeSegment.index].activeTab.index + 1 === Number(i)) {
        data[i].isActive = true;
      }
    }
    if (newSegments[activeSegment.index].activeTab.index + 1 === data.length) {
      switch (data.length) {
        case 1:
          data.push({
            name: 'location',
            title: 'Location',
            isActive: true,
            position: 'left',
          });
          break;
        case 2:
          data.push({
            name: 'sites',
            title: 'Sites',
            isActive: true,
            position: 'left',
          });
          break;
        case 3:
          data.push({
            name: 'preferredList',
            title: 'Preferred List',
            isActive: true,
            position: 'left',
          });
          break;
        default:
          null;
          break;
      }
      newSegments[activeSegment.index].activeTab = {
        name: data[data.length - 1].name,
        index: data.length - 1,
      };
    } else {
      newSegments[activeSegment.index].activeTab = {
        name: data[newSegments[activeSegment.index].activeTab.index + 1].name,
        index: newSegments[activeSegment.index].activeTab.index + 1,
      };
    }
    newSegments[activeSegment.index].tabs = data;

    dispatch(setSegments(newSegments));
  };

  return (
    <>
      <div className="feasibility-study-footer-container">
        <Button
          label={'Cancel'}
          size={'large'}
          handleButtonClick={handleCancel}
        />
        <Button
          className={'new-feasibility-footer-btn'}
          label={'Save and close'}
          size={'large'}
          handleButtonClick={() => handleSave()}
        />
        <Button
          handleButtonClick={
            segments[activeSegment.index]?.activeTab?.index === 3
              ? null
              : handleNext
          }
          disabled={
            (activeTab?.name === 'location' &&
              selectedCountries?.length === 0) ||
            (activeTab?.name === 'feasibility' &&
              patientCount?.feasibility === 0) ||
            (activeTab?.name === 'location' &&
              (!sites || Object.keys(sites)?.length === 0))
          }
          label={'Next'}
          size={'large'}
        />
      </div>
    </>
  );
};
export default FooterNewFeasibilityStudy;
