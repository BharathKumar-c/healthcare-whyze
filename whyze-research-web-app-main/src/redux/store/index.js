import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
