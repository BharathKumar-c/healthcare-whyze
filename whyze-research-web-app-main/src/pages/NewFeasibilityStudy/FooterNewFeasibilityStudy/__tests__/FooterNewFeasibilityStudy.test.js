import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../../redux/store';
import FooterNewFeasibilityStudy from '..';
import userEvent from '@testing-library/user-event';
import { SaveAndCloseModalConst } from '../../../../constant/ConstantTexts';
import { setSegments } from '../../../../redux/reducer/feasibilityStudyReducer';
import ApiUtil from '../../../../utils/ApiUtils';

const mockedUsedNavigate = jest.fn();

jest.mock('../../../../utils/ApiUtils');

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

function renderWithContext(element) {
  const { container, getByText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText };
}

function addTabsAndNextTabs(tabs, activeTab) {
  const newSegments = JSON.parse(
    JSON.stringify(store.getState().feasibilityStudyReducer.segments),
  );
  const { activeSegment } = JSON.parse(
    JSON.stringify(store.getState().feasibilityStudyReducer),
  );
  newSegments[activeSegment.index].activeTab = activeTab;
  newSegments[activeSegment.index].tabs = tabs;

  store.dispatch(setSegments(newSegments));
}

describe('Test cases for new feasibility study footer', () => {
  test('Create the snapshots for new feasibility study footer', () => {
    const { container } = renderWithContext(<FooterNewFeasibilityStudy />);

    expect(container).toMatchSnapshot();
  });

  test('Should render the close button correcly', () => {
    renderWithContext(<FooterNewFeasibilityStudy />);
    const buttons = screen.getAllByRole('button');
    const closeButton = buttons[0];
    expect(closeButton).toBeInTheDocument();
    expect(closeButton.textContent).toBe('Cancel');
  });

  test('Should render the save and close button correcly', () => {
    renderWithContext(<FooterNewFeasibilityStudy />);
    const buttons = screen.getAllByRole('button');
    const saveAndCloseButton = buttons[1];
    expect(saveAndCloseButton).toBeInTheDocument();
    expect(saveAndCloseButton.textContent).toBe('Save and close');
    userEvent.click(saveAndCloseButton);
    const whyzeLogo = screen.getByAltText('modal-title-logo');
    const titleText = screen.getByText(
      SaveAndCloseModalConst.SaveAndCloseModalTitle,
    );
    expect(whyzeLogo).toBeInTheDocument();
    expect(titleText).toBeInTheDocument();
  });

  test('Should render the next button correcly', () => {
    renderWithContext(<FooterNewFeasibilityStudy />);
    const buttons = screen.getAllByRole('button');
    const nextButton = buttons[2];
    expect(nextButton).toBeInTheDocument();
    expect(nextButton.textContent).toBe('Next');
  });

  test('Check the next button onclick function', () => {
    renderWithContext(<FooterNewFeasibilityStudy />);
    const buttons = screen.getAllByRole('button');
    const nextButton = buttons[2];

    userEvent.click(nextButton);
    const { segments, activeSegment } =
      store.getState().feasibilityStudyReducer;

    expect(segments[activeSegment.index]?.tabs[1]).toEqual({
      name: 'location',
      title: 'Location',
      isActive: true,
      position: 'left',
    });
  });

  test('Check the next button function when the next tab is already created', () => {
    const activeTab = {
      name: 'feasibility',
      index: 0,
    };
    const tabs = [
      {
        name: 'feasibility',
        title: 'Feasibility',
        isActive: true,
        position: 'left',
      },
      {
        name: 'location',
        title: 'Location',
        isActive: false,
        position: 'left',
      },
    ];
    addTabsAndNextTabs(tabs, activeTab);
    const { store } = renderWithContext(<FooterNewFeasibilityStudy />);

    const buttons = screen.getAllByRole('button');
    const nextButton = buttons[2];

    userEvent.click(nextButton);
    const { segments, activeSegment } = JSON.parse(
      JSON.stringify(store.getState().feasibilityStudyReducer),
    );

    expect(segments[activeSegment.index]?.tabs[1]).toEqual({
      name: 'location',
      title: 'Location',
      isActive: true,
      position: 'left',
    });
  });

  test('Should render the sites tab when click the next button in location tab', () => {
    const { store } = renderWithContext(<FooterNewFeasibilityStudy />);

    const buttons = screen.getAllByRole('button');
    const nextButton = buttons[2];
    userEvent.click(nextButton);

    const { segments, activeSegment } = JSON.parse(
      JSON.stringify(store.getState().feasibilityStudyReducer),
    );

    expect(segments[activeSegment.index]?.tabs[1]).toEqual({
      name: 'location',
      title: 'Location',
      isActive: false,
      position: 'left',
    });

    expect(segments[activeSegment.index]?.tabs[2]).toEqual({
      name: 'sites',
      title: 'Sites',
      isActive: true,
      position: 'left',
    });
  });

  test('Should render the preferred List tab when click the next button in sites tab', () => {
    const { store } = renderWithContext(<FooterNewFeasibilityStudy />);

    const buttons = screen.getAllByRole('button');
    const nextButton = buttons[2];
    userEvent.click(nextButton);

    const { segments, activeSegment } = JSON.parse(
      JSON.stringify(store.getState().feasibilityStudyReducer),
    );

    expect(segments[activeSegment.index]?.tabs[1]).toEqual({
      name: 'location',
      title: 'Location',
      isActive: false,
      position: 'left',
    });

    expect(segments[activeSegment.index]?.tabs[2]).toEqual({
      name: 'sites',
      title: 'Sites',
      isActive: false,
      position: 'left',
    });

    expect(segments[activeSegment.index]?.tabs[3]).toEqual({
      name: 'preferredList',
      title: 'PreferredList',
      isActive: true,
      position: 'left',
    });
  });

  test('Check the onClose fuction in save and close modal when click the x icon', async () => {
    renderWithContext(<FooterNewFeasibilityStudy />);

    const buttons = screen.getAllByRole('button');
    const saveAndCloseButton = buttons[1];
    expect(saveAndCloseButton).toBeInTheDocument();
    expect(saveAndCloseButton.textContent).toBe('Save and close');
    userEvent.click(saveAndCloseButton);

    const allButtons = screen.getAllByRole('button');

    const closeButton = allButtons[3];
    expect(closeButton).toBeInTheDocument();
  });

  test('Check the onClose fuction in save and close modal when click the cancel button', async () => {
    renderWithContext(<FooterNewFeasibilityStudy />);

    const buttons = screen.getAllByRole('button');
    const saveAndCloseButton = buttons[1];
    expect(saveAndCloseButton).toBeInTheDocument();
    expect(saveAndCloseButton.textContent).toBe('Save and close');
    userEvent.click(saveAndCloseButton);

    const allButtons = screen.getAllByRole('button');

    const cancelButton = allButtons[4];
    userEvent.click(cancelButton);
    expect(cancelButton).toBeInTheDocument();
  });

  test('Check the handleSave fuction in save and close modal when click the save button', async () => {
    renderWithContext(<FooterNewFeasibilityStudy />);

    ApiUtil.postData.mockResolvedValue({
      data: {
        message: 'Project stored successfully!',
      },
    });
    const buttons = screen.getAllByRole('button');
    const saveAndCloseButton = buttons[1];
    expect(saveAndCloseButton).toBeInTheDocument();
    expect(saveAndCloseButton.textContent).toBe('Save and close');
    userEvent.click(saveAndCloseButton);

    const inputFields = screen.getAllByRole('textbox');

    expect(inputFields[0]).toBeInTheDocument();
    expect(inputFields[0]).toHaveAttribute('name', 'projectName');
    userEvent.type(inputFields[0], 'test');

    expect(inputFields[1]).toBeInTheDocument();
    expect(inputFields[1]).toHaveAttribute('name', 'clientName');
    userEvent.type(inputFields[1], 'Sponser');

    expect(inputFields[2]).toBeInTheDocument();
    expect(inputFields[2]).toHaveAttribute('name', 'description');
    userEvent.type(inputFields[2], 'description');

    const projectInputField = screen.getByRole('spinbutton');
    fireEvent.change(projectInputField, { target: { value: 123 } });

    const allButtons = screen.getAllByRole('button');

    const saveButton = allButtons[5];
    expect(saveButton).toBeInTheDocument();

    userEvent.click(saveButton);

    await waitFor(() => {
      expect(ApiUtil.postData).toHaveBeenCalledTimes(1);
      expect(mockedUsedNavigate).toHaveBeenCalledWith('/dashboard');
    });
  });

  test('Check the handleSave fuction in save and close modal when click the save button', async () => {
    renderWithContext(<FooterNewFeasibilityStudy />);

    const buttons = screen.getAllByRole('button');
    const saveAndCloseButton = buttons[1];
    expect(saveAndCloseButton).toBeInTheDocument();
    expect(saveAndCloseButton.textContent).toBe('Save and close');
    userEvent.click(saveAndCloseButton);

    const inputFields = screen.getAllByRole('textbox');

    expect(inputFields[0]).toBeInTheDocument();
    expect(inputFields[0]).toHaveAttribute('name', 'projectName');
    fireEvent.change(inputFields[0], '');

    expect(inputFields[1]).toBeInTheDocument();
    expect(inputFields[1]).toHaveAttribute('name', 'clientName');
    userEvent.type(inputFields[1], 'Sponser');

    expect(inputFields[2]).toBeInTheDocument();
    expect(inputFields[2]).toHaveAttribute('name', 'description');
    userEvent.type(inputFields[2], 'description');

    const projectInputField = screen.getByRole('spinbutton');
    fireEvent.change(projectInputField, { target: { value: 123 } });

    const allButtons = screen.getAllByRole('button');

    const saveButton = allButtons[5];
    expect(saveButton).toBeInTheDocument();

    userEvent.click(saveButton);

    const projectInputErrorText = screen.getByText(
      'Please enter valid projectName',
    );

    expect(projectInputErrorText).toBeInTheDocument();
  });
});
