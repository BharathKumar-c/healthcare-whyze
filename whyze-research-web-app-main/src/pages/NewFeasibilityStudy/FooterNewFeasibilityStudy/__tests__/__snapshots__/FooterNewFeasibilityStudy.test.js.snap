// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for new feasibility study footer Create the snapshots for new feasibility study footer 1`] = `
<div>
  <div
    class="feasibility-study-footer-container"
  >
    <button
      class="ant-btn ant-btn-default ant-btn-lg"
      type="button"
    >
      <span>
        Cancel
      </span>
    </button>
    <button
      class="ant-btn ant-btn-default ant-btn-lg new-feasibility-footer-btn"
      data-testid="new-feasibility-footer-btn"
      type="button"
    >
      <span>
        Save and close
      </span>
    </button>
    <button
      class="ant-btn ant-btn-default ant-btn-lg"
      type="button"
    >
      <span>
        Next
      </span>
    </button>
  </div>
  <div />
</div>
`;
